src/coreclr/System.Private.CoreLib/src/Internal/Runtime/Binder/TextualIdentityParser.cs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Text;
+
+namespace Internal.Runtime.Binder
+{
+    internal static class TextualIdentityParser
```

Comment:
Should refactoring be separated since this PR is really big?

It's also worth tracking the usage of this formatter. If it's only used for diagnostics, we should not be concerned to change it.